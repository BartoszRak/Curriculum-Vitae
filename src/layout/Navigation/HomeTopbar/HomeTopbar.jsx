import React from 'react'
import PropTypes from 'prop-types'
import { Toolbar } from '@material-ui/core'
import { injectIntl } from 'react-intl'

import ResponsiveMenu from '~components/ResponsiveMenu'
import { useScrollTo, useScrollRangeHighlighting } from '~hooks'

import useStyle from './HomeTopbar.style'

export function HomeTopbar({ classes: overridingClasses, intl }) {
  const classes = { ...useStyle(), ...overridingClasses }
  const { formatMessage } = intl
  const { scrollTo } = useScrollTo(-300)
  const itemNames = ['commercial', 'projects', 'bio', 'skills', 'workflow', 'hobbies']

  //const { activeElement } = useScrollRangeHighlighting(itemNames)
  const items = itemNames.map(itemName => ({
    active: false, //Boolean(activeElement.id === itemName),
    name: formatMessage({ id: `layout.navigation.homeTopbar.items.${itemName}` }),
    props: {
      onClick: () => {
        const el = document.querySelector(`#${itemName}`)
        scrollTo(el)
      },
    },
  }))

  return (
    <Toolbar className={classes.toolbar}>
      <ResponsiveMenu items={items} />
    </Toolbar>
  )
}

HomeTopbar.propTypes = {
  classes: PropTypes.object,
  intl: PropTypes.object,
}

export default injectIntl(HomeTopbar)
