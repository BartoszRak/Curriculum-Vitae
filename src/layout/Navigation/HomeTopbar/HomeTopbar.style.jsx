import { makeStyles } from '@material-ui/core'

export default makeStyles(() => ({
  toolbar: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    width: '100%',
  },
}))
