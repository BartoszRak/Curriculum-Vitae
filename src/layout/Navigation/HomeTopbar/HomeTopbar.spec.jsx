import React from 'react'
import { cleanup } from '@testing-library/react'
import { shallow } from 'enzyme'

import { HomeTopbar } from './HomeTopbar'

jest.unmock('@material-ui/core')

jest.mock('./Menu', () => 'MenuMock')

let props

describe('component: HomeTopbar', () => {
  afterEach(cleanup)

  beforeEach(() => {
    props = {
      intl: {
        formatMessage: jest.fn(),
      },
    }
  })
  describe('rendering', () => {
    test('render without crush', () => {
      const wrapper = shallow(<HomeTopbar {...props} />)

      expect(wrapper).toBeTruthy()
    })

    test('match snapshot ', () => {
      const wrapper = shallow(<HomeTopbar {...props} />)

      expect(wrapper).toMatchSnapshot()
    })

    test('render menu', () => {
      const wrapper = shallow(<HomeTopbar {...props} />)

      expect(wrapper.find('MenuMock').length).toBe(1)
    })
  })
})
