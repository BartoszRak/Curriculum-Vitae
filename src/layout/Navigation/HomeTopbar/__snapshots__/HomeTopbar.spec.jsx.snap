// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: HomeTopbar rendering match snapshot  1`] = `
<WithStyles(ForwardRef(Toolbar))
  className="makeStyles-toolbar-1"
>
  <ResponsiveMenuMock
    items={
      Array [
        Object {
          "name": undefined,
          "props": Object {
            "active": false,
            "onClick": [Function],
          },
        },
        Object {
          "name": undefined,
          "props": Object {
            "active": false,
            "onClick": [Function],
          },
        },
        Object {
          "name": undefined,
          "props": Object {
            "active": false,
            "onClick": [Function],
          },
        },
        Object {
          "name": undefined,
          "props": Object {
            "active": false,
            "onClick": [Function],
          },
        },
        Object {
          "name": undefined,
          "props": Object {
            "active": false,
            "onClick": [Function],
          },
        },
        Object {
          "name": undefined,
          "props": Object {
            "active": false,
            "onClick": [Function],
          },
        },
      ]
    }
  />
</WithStyles(ForwardRef(Toolbar))>
`;
